src/main/java/org/prebid/server/auction/ExchangeService.java

```diff
@@ -672,20 +673,47 @@ private static ExtPriceGranularity parsePriceGranularity(JsonNode priceGranulari
      * <p>
      * Returns input argument as the result if no errors found or create new {@link BidderSeatBid} otherwise.
      */
-    private BidderSeatBid validateAndUpdateResponse(BidderSeatBid bidderSeatBid) {
+    private BidderSeatBid validateAndUpdateResponse(BidderSeatBid bidderSeatBid, List<String> requestCurrencies) {
+        if (requestCurrencies.size() == 0) {
```

Comment:
Please extract seatbid currency validation and bids validation into separate methods which `validateAndUpdateResponse` will call.